common/src/main/java/tachyon/conf/TachyonConf.java

```diff
@@ -229,16 +229,35 @@ public void set(String key, String value) {
   }
 
   public String get(String key, final String defaultValue) {
-    String raw = mProperties.getProperty(key, defaultValue);
-    String updated = lookup(raw);
-    LOG.debug("Get Tachyon property {} as {} with default {}", key, updated, defaultValue);
-    return updated;
+    if (mProperties.containsKey(key)) {
+      String raw = mProperties.getProperty(key);
+      String updated = lookup(raw);
+      LOG.debug("Get Tachyon property {} as {} with default {}", key, updated, defaultValue);
+      return updated;
+    }
+    return defaultValue;
+  }
+
+  public int get(String key, final int defaultValue) {
+    if (mProperties.containsKey(key)) {
+      String raw = mProperties.getProperty(key);
+      String updated = lookup(raw);
```

Comment:
NIT: calculating updated part shows up in more than one places. Consider abstracting it out when you work on the TODO at line 259